### PR TITLE
API-1025: Rework collect of Audit data_source events

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/CountDailyEventsByConnectionHandler.php [hunk 1]
```diff
@@ -22,53 +22,25 @@ public function __construct(SelectConnectionsEventCountByDayQuery $selectConnect
...
             ->selectConnectionsEventCountByDayQuery
-            ->execute($query->eventType(), $fromUtcDateTime, $upToUtcDateTime);
+            ->execute($query->eventType(), $query->fromDateTime(), $query->upToDateTime());
 
-        $weeklyEventCounts = [];
-        foreach ($hourlyEventsPerConnection as $connectionCode => $hourlyEventCounts) {
-            $weeklyEventCounts[] = new WeeklyEventCounts(
+        $periodEventCounts = [];
+        foreach ($hourlyEventCountsPerConnection as $connectionCode => $hourlyEventCounts) {
+            $periodEventCounts[] = new PeriodEventCount(
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/CountDailyEventsByConnectionHandler.php:36]: `PeriodEventCount` instantiation could be move inside the Query

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Persistence/Query/SelectConnectionsEventCountByDayQuery.php [hunk 2]
```diff
@@ -12,13 +12,13 @@
 interface SelectConnectionsEventCountByDayQuery
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Domain/Audit/Persistence/Query/SelectConnectionsEventCountByDayQuery.php:12]: Could be rename to `SelectPeriodEventCounts` or `SelectPeriodEventCountByConnection`

File: src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/CountDailyEventsByConnectionHandler.php (hunk 1 above)
Comment (fitn) [reply src/Akeneo/Connectivity/Connection/back/Application/Audit/Query/CountDailyEventsByConnectionHandler.php:36]: Yes, I'd also say that it should be move :P
It is one of the objectives of the rework to make the db query objects homogeneous with the existing stack

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/PeriodEventCount.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    private $fromDateTime;
+
+    /** @var \DateTimeImmutable */
+    private $upToDateTime;
+
+    /** @var HourlyEventCount[] */
+    private $hourlyEventCounts;
+
+    /**
+     * @param HourlyEventCount[] $hourlyEventCounts
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/PeriodEventCount.php:29]: If you add php docs, do it for all params pls.

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalSelectPeriodEventCountsQuery.php [hunk 4]
```diff
@@ -109,31 +115,24 @@ private function getHourlyEventCountsForAllConnections(
...
-     * Example:
-     * [
-     *   'erp' => [
-     *     [DateTime(2020-01-01 00:00:00), 1],
-     *     [DateTime(2020-01-01 01:00:00), 3],
-     *     ...
-     *     [DateTime(2020-01-09 00:00:00), 7]
-     *   ],
-     *   ...
+     * @param array $hourlyEventCountsData = [
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalSelectPeriodEventCountsQuery.php:122]: same

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/PeriodEventCount.php (hunk 3 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/PeriodEventCount.php:29]: It's not necessary if you add no valuable information, the editor/IDE will infer the type from the code.

**Pull request merged**